core/src/main/java/com/backblaze/b2/client/contentSources/B2Headers.java

```diff
@@ -85,6 +92,42 @@ default long getContentLength() {
         }
     }
 
+    /**
+     * @return the value of the X-Bz-File-Name header (percentDecoded)
+     *         or null if that header isn't present or can't be percentDecoded.
+     * @apiNote We return null here because this is a completely optional, non-standard header
+     *          and anyone could put any type of value in it at any time.
+     */
+    default String getFileNameOrNull() {
+        final String str = getValueOrNull(B2Headers.FILE_NAME);
+        if (str == null) {
+            return null;
+        }
+
+        try {
+            return percentDecode(str);
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    /**
+     * @return the value of the X-Bz-Upload-Timestamp header (parsed as a long)
+     *         or null if that header isn't present or can't be parsed.
+     * @apiNote We return null here because this is a completely optional, non-standard header
+     *          and anyone could put any type of value in it at any time.
+     */
+    default Long getUploadTimestampOrNull() {
+        final String str = getValueOrNull(B2Headers.UPLOAD_TIMESTAMP);
+        B2Preconditions.checkState(str != null, "don't call if there isn't a b2 filename");
```

Comment:
wrong error message.

Reply:
thanks.  it's even worse than that.  the comment says we'll return null if it's not present and the code is throwing an exception.  i've added tests and updated the code.